extension/socks5proxy/bin/winfwpolicy.cpp

```diff
@@ -0,0 +1,308 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "winfwpolicy.h"
+
+#include <WS2tcpip.h>
+#include <fwpmu.h>
+#include <iptypes.h>
+
+#include <QScopeGuard>
+#include <QSettings>
+#include <QString>
+#include <QUuid>
+
+#include "winutils.h"
+
+#pragma comment(lib, "Fwpuclnt")
+
+// Fixed GUID of the VPN Killswitch firewall sublayer
+constexpr const QUuid KILLSWITCH_FW_GUID(0xc78056ff, 0x2bc1, 0x4211, 0xaa, 0xdd,
+                                         0x7f, 0x35, 0x8d, 0xef, 0x20, 0x2d);
+
+// Fixed GUID of our own firewall sublayer
+constexpr const QUuid LOCALPROXY_FW_GUID(0x0555706c, 0x4468, 0x4ec6, 0xb4, 47,
+                                         0x20, 0xe7, 0x4a, 0x10, 0x06, 0xe7);
+
+WinFwPolicy::WinFwPolicy(QObject* parent) : QObject(parent) {
```

Comment:
Totally nitpicking here. Maybe a factory function, given the constructor can fail.